src/eventsource.js

```diff
@@ -929,10 +935,11 @@
             abortController = undefined;
           }
         } else {
+          var nextHeartbeat = Math.max((wasActivity || Date.now()) + heartbeatTimeout - Date.now(), 1);
```

Comment:
shouldn't this be `Math.max(Math.min(Date.now() - wasActivity, heartbeatTimeout), 1000)` ?